### PR TITLE
docs(dev): Add initial high-level document for CI

### PR DESCRIPTION
**Pull request by cradkowski**
I added the initial high-level description of what CI is and included basic instructions for enabling this in GitHub and GitLab (eventually will further link to more specific documents for each platform later). 
Any edits or suggestions are welcome, particularly for my definition of what Continuous Integration is. 

Signed-off-by: Chris Radkowski <cradkowski@worcester.edu>

### CODE REVIEW

**Review by StoneyJackson — COMMENTED** [review]
@cradkowski Great start! I'm going to make some revisions and add CD. I'll ask you to review when I'm done.

**Comment by StoneyJackson** [comment]
@cradkowski hmm... I can't add you as a reviewer. Consider yourself added. If you are good with it, I'll merge. If not, we'll make more changes :)

File: docs/dev/continuous-integration-and-deployment.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Continuous Integration and Deployment/Delivery
+
+For a more in-depth look at CI/CD see [GitHub's video about CI](https://youtu.be/xSv_m3KhUO8)
```
Comment (cradkowski) [inline docs/dev/continuous-integration-and-deployment.md:3]: Maybe change the link name since the video is about CI and CD.

**Review by cradkowski — COMMENTED** [review]
I approve these changes.

**Pull request closed without merging**